includes/spam.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Code related to spam detection and prevention.
+ */
+// Constants. Define these in wp-config.php to override.
+if ( ! defined( 'PMPRO_SPAM_ACTION_NUM_LIMIT' ) ) {
+    define( 'PMPRO_SPAM_ACTION_NUM_LIMIT', 10 );
+}
+if ( ! defined( 'PMPRO_SPAM_ACTION_TIME_LIMIT' ) ) {
+    define( 'PMPRO_SPAM_ACTION_TIME_LIMIT', 900 );  // in seconds
+}
+
+/**
+ * Is the current visitor a spammer?
+ * @since 2.7
+ */
+function pmpro_is_spammer() {
+    $is_spammer = false;
+    
+    $activity = pmpro_get_spam_activity();
+    if ( count( $activity ) >= PMPRO_SPAM_ACTION_NUM_LIMIT ) {

```

Comment:
Exclude `false` if we could not determine IP.

```suggestion
    if ( false !== $activity && count( $activity ) >= PMPRO_SPAM_ACTION_NUM_LIMIT ) {

```